haystack/core/component/component.py

```diff
@@ -160,6 +160,20 @@ def __call__(cls, *args, **kwargs):
         return instance
 
 
+def _component_repr(component: Component) -> str:
+    """
+    All Components override their __repr__ method with this one.
+    It prints the component name and the input/output sockets.
+    """
+    # This is similar to a default __repr__ method but doesn't include the full module
```

Comment:
What's the downside of displaying the full module path? I'd argue that it's more helpful to show it all.

Reply:
If it's deeply nested it could be a reaaaally long string, but I agree it's much more clear if shown in full. I'll change it.